### ISSUE TITLE
Can't create file or directory in Cosmos FileSystem.

### ISSUE THREAD

**Issue by triploit** [CosmosOS/Cosmos#491]
Hey everyone!

I can't create a new dirextory or a new file, whether i use `System.IO` or `Cosmos.System.FileSystem.VFS`. I can read from a file, but i can't create or write into it.

Is it possible, that that's a bug or is it releated to my code? I've initialized the filesystem with this code:

`private static Sys.FileSystem.CosmosVFS FS;`

`FS = new Sys.FileSystem.CosmosVFS();
Sys.FileSystem.VFS.VFSManager.RegisterVFS(FS);
FS.Initialize();`

And so I create a folder:

`Directory.CreateDirectory("0:\\TestFolder");`

Is that right or have I to do something else to make it work?
With kind regards, Survari.

(Please do not be too strict with me, I'm from Germany and the english grammar is not my strength)

**Comment by jp2masa** [comment]
It's not working at the moment. You can try an older devkit or wait that it gets fixed.

**Comment by triploit** [comment]
Ok, thanks :). I will try an older one.

**Comment by ROBLOXVIP735** [comment]
I don't know. Honestly, not sure HOW to download older versions.

**Comment by jp2masa** [comment]
If you are having troubles, you can download an unofficial compiled version of Cosmos in http://jp2masa.tk

**Comment by ROBLOXVIP735** [comment]
Thanks. Is this complete with an old version?

**Comment by jp2masa** [comment]
Complete like what?

**Comment by ROBLOXVIP735** [comment]
I meant did the site support old versions, which it does. However, for some reason, I cannot open the .zip files.

**Comment by jp2masa** [comment]
The compiled Cosmos versions in that website are a clone of the master branch on the date that is written there. Also, in Windows you can usually open .zip files in the Explorer.

**Comment by ROBLOXVIP735** [comment]
I use 7 zip, but none of them except the oldest work. I used explorer too.

**Comment by cjhannah** [comment]
You should ptu this in a new issue.

**Comment by ROBLOXVIP735** [comment]
Will do.

**Comment by charlesbetros** [comment]
I think this is already logged as an issue...

**Comment by denisbest** [comment]
Hi, I also want to make OS with COSMOS. Only thing I need to figure out, is how to make a simple file system. Can you give me some tips? I need to create file, create folder, that stuff... COSMOS FAQ is very poor about file creation.

I thought doing it with vectors/arrays but when OS restart, everything will be erased and I don't want that. I need permanent solution.

**Comment by zarlo** [comment]
FAT (the file system we use) has a lot of bugs and is new, you can try and fix the bugs it has

**Comment by jp2masa** [comment]
First, you need the devkit, but in the most recent devkits filesystems aren't working, so you'll need a devkit from maybe 3 months ago. Then just use usual System.IO methods with drive 0:\

**Comment by denisbest** [comment]
I already have recent devkit, do I need to uninstall it first or just run installation of old devkit?

**Comment by jp2masa** [comment]
Just run installation of old devkit.

**Comment by denisbest** [comment]
jp2masa, can you please make me and for everyone else with same problem, a tutorial how to create files and directory's? I installed Cosmos User Kit - 19/07/2016 - non-official

**Comment by jp2masa** [comment]
If you're running it in a VM directly from Visual Studio, you have a partition mounted as 0:\ so you just need to do like:
```
System.IO.Directory.CreateDirectory(@"0:\MyDir");
```
or
```
System.IO.File.Create(@"0:\MyFile.txt");
```

There's a bug that doesn't allow to write in files inside directories. I'm now working on trying to fix the directory and file creation and after that I'll try to fix that bug.

**Comment by denisbest** [comment]
Can I add you to skype, slack, steam or something like that?

I did that and it worked:

private static Sys.FileSystem.CosmosVFS FS;
FS = new Sys.FileSystem.CosmosVFS(); Sys.FileSystem.VFS.VFSManager.RegisterVFS(FS); FS.Initialize();
FS.CreateDirectory("0:\\TestDir");
FS.CreateFile("0:\\TestDir\\testFile.txt");

But how can I write to file?

**Comment by jp2masa** [comment]
File.WriteAllText
You can join Cosmos gitter or talk with me on gitter too. https://gitter.im/CosmosOS/Cosmos

**Comment by charlesbetros** [comment]
Fixed by 1c8901850cd33577b5a69b04d6c990e948134018
